### PR TITLE
feat(tablecardeditform): allow edit form to be better customized

### PR DESCRIPTION
**Pull request by scottdickerson**
Closes #

**Summary**

- The TableCard needs to have additional abilities to edit the CardEditor panels to support our usecases

**Change List (commits, features, bugs, etc)**

- feat(CardEditFormContent): attach a class so styles can be more controlled and allow title, tooltip to be optional as well as ContentFormItemTitle
- feat(ContentFormItemTitle): allow title and tooltip to be optional
- feat(TableCardFormContent): attach a class so styles can be more controlled
- feat(card-edit-form.scss): remove incorrect --content styles (was not attached originally) and set to flex so it can be more controlled, same with the new table-card-form-content class
- feat(CardEditor): add new callback to dynamically provide valid dimensions instead of hardcoding them dashboard wide
- test(CardEditor.story): add an example of the table card being customized

**Acceptance Test (how to verify the PR)**

- Verify the new table card story of card editor correctly shows dimension values and updates the card when the DataType is changed

**Regression Test (how to make sure this PR doesn't break old functionality)**

- Verify the other CardEditor stories still match the old ones (no snapshot changes)

**Things to look for during review**

- [ ] Make sure all references to `iot` or `bx` class prefix is using the prefix variable
- [ ] (React) All major areas have a `data-testid` attribute. New test ids should have test written to ensure they are not changed or removed.
- [ ] UI should be checked in RTL mode to ensure the proper handling of layout and text.
- [ ] All strings should be translatable.
- [ ] The code should pass a11y scans (The storybook a11y knob should show no violations). New components should have a11y test written.
- [ ] Unit test should be written and should have a coverage of 90% or higher in all areas.
- [ ] All components should be passing visual regression test. For new styles or components either a visual regression test should be written for all permutations or the base image updated.
- [ ] Changes or new components should either write new or update existing documentation.
- [ ] PR should link and close out an existing issue

### CODE REVIEW

**Comment by bjornalm** [comment]
> Verify the new table card story of card editor correctly shows dimension values and updates the card when the DataType is changed
>

I'm testing [this story](https://deploy-preview-2654--carbon-addons-iot-react.netlify.app/?path=/story/2-watson-iot-experimental-%E2%98%A2%EF%B8%8F-cardeditor--table-card-with-custom) but I'm not 100% sure what to expect. Changing the "Data source" does not update the component, but it does trigger 3 callbacks. Can you be a bit more explicit on what we need to look for?

**Comment by scottdickerson** [comment]
@bjornalm yes the 3 callbacks should be triggered with the updated card dataType as you change the dataType.  That's the end to end case

**Review by kevinsperrine — APPROVED** [review]

File: packages/react/src/components/CardEditor/CardEditForm/CardEditFormContent.jsx [hunk 1]
```diff
@@ -209,14 +212,17 @@ const CardEditFormContent = ({
...
-        ? editContentSections.map(({ header: { title, tooltip }, content }) => (
-            <React.Fragment key="custom-content-section">
-              <ContentFormItemTitle title={title} tooltip={tooltip} />
-              {content}
-            </React.Fragment>
-          ))
+        ? editContentSections.map(({ header, content }) => {
+            const { title, tooltip } = header || {};
+            return (
+              <React.Fragment key="custom-content-section">
```
[the remark this answers is not in the document]
Comment (davidicus) [reply packages/react/src/components/CardEditor/CardEditForm/CardEditFormContent.jsx:218]: Is there a state issue by giving each one the same key value?

[the remark this answers is not in the document]
Comment (scottdickerson) [reply packages/react/src/components/CardEditor/CardEditForm/CardEditFormContent.jsx:218]: thanks, fixed this in my latest

**Review by davidicus — APPROVED** [review]

**Pull request merged**